hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/cache/CacheAddInvalidationListenerTask.java

```diff
@@ -46,13 +51,24 @@ public void handleEvent(Object eventObject) {
                     return;
                 }
                 if (eventObject instanceof CacheInvalidationMessage) {
-                    CacheInvalidationMessage message = (CacheInvalidationMessage) eventObject;
-
-                    ClientMessage eventMessage
-                            = CacheAddInvalidationListenerCodec.encodeCacheInvalidationEvent(message.getName(),
-                            message.getKey(), message.getSourceUuid());
-                    sendClientMessage(message.getKey(), eventMessage);
-
+                    if (eventObject instanceof CacheSingleInvalidationMessage) {
+                        CacheSingleInvalidationMessage message = (CacheSingleInvalidationMessage) eventObject;
+                        ClientMessage eventMessage = CacheAddInvalidationListenerCodec.
+                                    encodeCacheInvalidationEvent(message.getName(), message.getKey(), message.getSourceUuid());
+                        sendClientMessage(message.getName(), eventMessage);
```

Comment:
- to be consistent with batch event
- `key` may be `null` sometimes such as invalidates all entries (clear near-cache) when cache clear
- ordering events as their keys doesn't make sense to me